### PR TITLE
Fix NPE in BreadcrumbsImpl, Update TextMate library, update typescript grammer and language server

### PR DESCRIPTION
**Pull request by matthiasblaesing**

### CODE REVIEW

File: ide/textmate.lexer/external/org.eclipse.tm4e.core-0.4.1-pack1-license.txt [hunk 1]
```diff
@@ -0,0 +1,92 @@
+Name: TextMate support for Eclipse
+Version: 0.4.1-pack1
+License: EPL-v20
+Description: Used to interpret TextMate grammars
+Origin: https://github.com/eclipse/tm4e
+
+Eclipse Public License - v 2.0
```
Comment (sdedic) [inline ide/textmate.lexer/external/org.eclipse.tm4e.core-0.4.1-pack1-license.txt:7]: Possible mimatch: https://github.com/eclipse/tm4e/blob/master/LICENSE states EPL 1.0 even in the master version.

File: webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java [hunk 2]
```diff
@@ -163,27 +166,32 @@ public static void main(String... args) throws IOException, InterruptedException
...
-        
+
+        try(Writer binariesList = new OutputStreamWriter(Files.newOutputStream(bundlesDir.resolve("binaries-list")), "UTF-8")) {
+            binariesList.write(readResourceIntoString("ALv2Header.txt"));
+            for(Entry<String,String> e: binaries.entrySet()) {
+                binariesList.write(e.getValue() + " " + e.getKey() + nl);
+            }
+        }
+
         Map<String, String> project2LicenseKey = new HashMap<>();
```
Comment (sdedic) [inline webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java:187]: Consider converting to `TreeMap`, too as the `-license.txt` is created in entrySet-iteration order.

File: ide/textmate.lexer/external/org.eclipse.tm4e.core-0.4.1-pack1-license.txt (hunk 1 above)
Comment (matthiasblaesing) [reply ide/textmate.lexer/external/org.eclipse.tm4e.core-0.4.1-pack1-license.txt:7]: Thank you for checking this. However I checked a few of the files in the core project, and I see EPL-v2.0 - checked:

https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/registry/Registry.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/internal/grammar/Grammar.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/internal/grammar/parser/Raw.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/internal/grammar/reader/SyncGrammarReader.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/internal/parser/xml/XMLPListParser.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/grammar/IGrammar.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/model/Tokenizer.java
https://github.com/eclipse/tm4e/blob/master/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/theme/RGB.java
https://github.com/eclipse/tm4e/blob/master/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/theme/css/CSSStyle.java
https://github.com/eclipse/tm4e/blob/0.4.1/org.eclipse.tm4e.core/src/main/java/org/eclipse/tm4e/core/model/DecodeMap.java

File: ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/BreadcrumbsImpl.java [hunk 3]
```diff
@@ -104,33 +102,31 @@ private DocumentSymbol toDocumentSymbol(Either<SymbolInformation, DocumentSymbol
...
             return ;
         }
 
-        element = ((RootBreadcrumbsElementImpl) element).children.get(0);
-        //TODO: stale results... modified while the query is running?
+        if (comp.getCaret() != null && (!element.getChildren().isEmpty())) {
+            element = element.getChildren().get(0);
 
-        int caret = comp.getCaretPosition();
+            int caret = comp.getCaretPosition();
```
[the remark this answers is not in the document]
Comment (matthiasblaesing) [reply ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/BreadcrumbsImpl.java]: Valid point - updated PR accordingly. Thank you.

File: webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java (hunk 2 above)
Comment (matthiasblaesing) [reply webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java:187]: These files are independent from each other, so creation order of the license files does not matter for minimizing the diff. The order only matters in the binaries-list. But maybe I'm overlooking something?

Comment (sdedic) [reply webcommon/typescript.editor/bundles/prepare/src/main/java/org/netbeans/modules/learning/prepare/PrepareBundles.java:187]: Nothing overly important, just a predictable ordering of the output.

**Review by sdedic — APPROVED** [review]

**Review by jlahoda — APPROVED** [review]
Looks OK.

**Comment by matthiasblaesing** [comment]
@sdedic @jlahoda thank you for review.

**Pull request merged**